### PR TITLE
Add output before saving image during export to indicate progress

### PR DESCRIPTION
**Pull request by jabrown85**
Resolves: https://github.com/buildpacks/lifecycle/issues/500

Signed-off-by: Jesse Brown <jabrown85@gmail.com>

### CODE REVIEW

File: save.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ import (
 func saveImage(image imgutil.Image, additionalNames []string, logger Logger) (platform.ImageReport, error) {
 	var saveErr error
 	imageReport := platform.ImageReport{}
+	logger.Infof("Publishing %s\n", image.Name())
```
Comment (jabrown85) [inline save.go:18]: I thought about listing all the image names like we do in the `*** Images` section. I also thought about a more generic message like `Pushing image(s)...`. Let me know what you think!

Comment (yaelharel) [reply save.go:18]: Maybe it's better that we'll print the same value as we print in the following line: https://github.com/buildpacks/lifecycle/blob/203221553c114ecbad7378e16ff02385dd86d23d/save.go#L34
I think that printing `image.Name()` is clearer but maybe there is a reason that we print `shortID(id)` in the other place.

Comment (jabrown85) [reply save.go:18]: Will the `id` be available via `image.Identifier()` _before_ the push? I thought that was likely invalid to use until a push to a registry had occurred.

Comment (yaelharel) [reply save.go:18]: ohhhh interesting. I'm not sure :\
If it's not, maybe you can just add `image.Name` to the print after the push (in addition to `shortID(id)`).
I think that in this way it'll be easier to the reader to see the progress.

Comment (jabrown85) [reply save.go:18]: The output already has the name it looks like
```
Publishing 192.168.86.20:5005/myrepo/img **new line
*** Images (sha256:cb66dc1599dd7be223b5dc3e88dc9f2eef1aec0aaf609c56aef2cb3ee394ff84):
      192.168.86.20:5005/myrepo/img
```

Comment (jabrown85) [reply save.go:18]: All the tags are output for the image that was pushed, so I think it lines up ok.

Comment (yaelharel) [reply save.go:18]: You're right. I missed that we're printing more information after https://github.com/buildpacks/lifecycle/blob/203221553c114ecbad7378e16ff02385dd86d23d/save.go#L34
Thanks for the clarification!

**Review by yaelharel — APPROVED** [review]

**Review by ekcasey — APPROVED** [review]

File: save.go [hunk 2]
```diff
@@ -15,6 +15,7 @@ import (
 func saveImage(image imgutil.Image, additionalNames []string, logger Logger) (platform.ImageReport, error) {
 	var saveErr error
 	imageReport := platform.ImageReport{}
+	logger.Infof("Saving %s\n", image.Name())
```
Comment (natalieparellano) [inline save.go:18]:
```suggestion
	logger.Infof("Saving %s...\n", image.Name())
```

Comment (natalieparellano) [reply save.go:18]: IDK if this is really needed it is just a preference. Having tried a build with this branch out, I must say the message when there's a pause is psychologically satisfying :)

**Review by natalieparellano — APPROVED** [review]

File: save.go (hunk 2 above)
Comment (natalieparellano) [reply save.go:18]: Actually there's no `...` on the other "Doing XXX" log messages so what we have now is probably more consistent.

Comment (natalieparellano) [reply save.go]: @jabrown85 I'm sorry... I made this comment and then tried to take it back :/ I am totally fine with whatever we decide to do :)

Comment (jabrown85) [reply save.go]: I liked the `...` suggestion. But if someone feels strongly that we shouldn't introduce that, I'll back it out.

**Pull request merged**